src/api/core/organizations.rs

```diff
@@ -2849,6 +2850,19 @@ async fn put_reset_password_enrollment(
         err!("Reset password can't be withdrawed due to an enterprise policy");
     }
 
+    let user = headers.user;
+
+    if reset_request.ResetPasswordKey.is_some() {
+        match reset_request.MasterPasswordHash {
+            Some(password) => {
+                if !user.check_valid_password(&password) {
```

Comment:
No need to define `user` here as a new variable.
You can just use:

```suggestion
    if reset_request.ResetPasswordKey.is_some() {
        match reset_request.MasterPasswordHash {
            Some(password) => {
                if !headers.user.check_valid_password(&password) {
```